framework/src/binary_manager/binary_manager_interface.c

```diff
@@ -24,10 +24,23 @@
 #include <errno.h>
 #include <debug.h>
 #include <sched.h>
+#include <unistd.h>
 #include <fcntl.h>
 #include <mqueue.h>
 #include <tinyara/binary_manager.h>
 
+void binary_manager_set_request(binmgr_request_t *request_msg, int cmd, char *bin_name, binmgr_cb_t *callback)
+{
+	request_msg->cmd = cmd;
+	request_msg->requester_pid = getpid();
+	if (bin_name != NULL) {
+		snprintf(request_msg->data.bin_name, BIN_NAME_MAX, "%s", bin_name);
```

Comment:
request_msg->data.bin_name and cb_info is union value. Please check that this is correct or not.